python/ray/serve/api.py

```diff
@@ -232,6 +238,23 @@ def create_backend(
 
         if config is None:
             config = {}
+        if ray_actor_options is None:
+            ray_actor_options = {}
+        if env is None:
+            # If conda is activated, default to conda env of the driver.
+            if os.environ.get("CONDA_PREFIX"):
+                ray_actor_options.update(override_environment_variables={
+                    "PYTHONHOME": os.environ.get("CONDA_PREFIX")
+                })
+        else:
+            if not os.environ.get("CONDA_PREFIX"):
+                raise ValueError("conda must be activated to use CondaEnv.")
+            python_home = os.path.join(
+                os.path.split(os.environ.get("CONDA_PREFIX"))[0], env.name)
+            if not os.path.isdir(python_home):
+                raise ValueError("conda env " + env.name + " not found.")
+            ray_actor_options.update(
+                override_environment_variables={"PYTHONHOME": python_home})
```

Comment:
Unlikely to be an issue, but we should have a check if `override_environment_variables` is already set by the user and if it is, just update it instead of totally overriding it.

Reply:
Maybe we need to make the override_environment_variables notation more clear--it has the same semantics as `dict.update()`, so it keeps earlier overrides the same and updates PYTHONHOME.